src/cargo/core/resolver/features.rs

```diff
@@ -177,6 +177,10 @@ impl FeatureOpts {
         if let ForceAllTargets::Yes = force_all_targets {
             opts.ignore_inactive_targets = false;
         }
+        if unstable_flags.weak_dep_features {
+            // Force this ON because it only works with the new resolver.
+            opts.new_resolver = true;
```

Comment:
The new resolver is intended to be 100% identical to the old one, assuming none of the other options are enabled (like `decouple_host_deps`).  The way I viewed this is that this is just an internal implementation detail that should be invisible and irrelevant to users.  I considered using `-Z features=weak` as the CLI interface, but the parsed flags aren't available from `Config` or `Summary` where they are needed.  I could special-case it, but it seemed simpler just to have a single separate flag.  

The long-term stabilization plan here is that we just turn this on for everyone (set `new_resolver` and `weak_dep_features` to TRUE by default, and eventually just remove them).  There will be no opt-in to it, since it should be backwards compatible.  The other resolver options will remain tied to the `resolver = "2"` opt-in, but technically that will just turn on the other options (`decouple_host_deps` and friends).

Does that make sense?  It does seem a bit confusing from the perspective of how it is implemented, but from the perspective of the user, it is more about opting-in to backwards-incompatible changes.  The `resolver` opt-in is about the user's perception of things being different, and less about how it is actually implemented.

Reply:
Oh right sorry I keep equating "new resolver" with all the new features (like `decouple_host_deps`). I forgot that the new resolver also implements the old logic! In that case this seems fine and agreed this should be a silent switch!